Add guides for Navigation of screens in NUI

jaehyun0cho opened the pull request:
NUI provides a system to organize screens and navigate between the screens by using Page and Navigator.

To organize a formatted full screen page, ContentPage is used. To show a dialog on the screen, DialogPage is used.

Navigator pushes and pops pages to its LIFO structure. The peek page, which has been pushed last, is displayed on the screen by overlapping the previous page.

### Change Description ###

Describe your changes here.

For instance,
 - Fix broken links of native application docs.
 -  ...

### Bugs Fixed ###

Provide links to bugs here

For instance,

 - Issue #265

### API Changes ###

In case of ACR, put the link of ACR

For instance,

 - ACR: ACR-1120

omar-srbd commented on docs/application/dotnet/guides/user-interface/nui/nui-components/navigation/DialogPage.md:
```diff
@@ -0,0 +1,165 @@
+
+[DialogPage](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.DialogPage.html) helps to show a dialog on the screen. `DialogPage` is composed of [Content](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.DialogPage.html#Tizen_NUI_Components_DialogPage_Content) and [Scrim](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.DialogPage.html#Tizen_NUI_Compon ...
+
+`Content` property indicates the region to show the dialog on the screen. [AlertDialog](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.AlertDialog.html) or [Dialog](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.Dialog.html) is set to `Content` property.
```
```suggestion
`Content` property indicates the region to show the dialog on the screen. [Tizen.NUI.Components.AlertDialog](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.AlertDialog.html) class or [Tizen.NUI.Components.Dialog](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.Dialog.html) class is set to `Content` property.
```

omar-srbd commented on docs/application/dotnet/guides/user-interface/nui/nui-components/navigation/DialogPage.md:
```diff
@@ -0,0 +1,165 @@
+
+## Related information
+
+- See also [AlertDialog](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.AlertDialog.html).
+
+- See also [Dialog](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.Dialog.html).
+
+- See also [Page](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.Page.html).
+
+- See also [Navigator](/application/dotnet/api/TizenFX/latest/api/Tizen.NUI.Components.Navigator.html).
+
```
```suggestion
```

omar-srbd reviewed (commented):
Hello, @jaehyun0cho
I have reviewed the files and left some correction suggestions. Please incorporate them at your earliest convenience.

Thanks

jaehyun0cho commented:
The files have been updated based on the review comments.
Thank you.

jaehyun0cho commented on docs/application/dotnet/guides/user-interface/nui/nui-components/navigation/navigation.md:
```diff
@@ -0,0 +1,35 @@
+    }
+</style>
+
+# Navigation of screens
```
As you suggested on messenger, I changed the file name from navigation to Navigator.
So I changed the title of that page from Navigation of screens to Navigator.
Thank you.

omar-srbd reviewed (commented):
@jaehyun0cho, thanks for your most recent commit incorporating changes from my suggestions.
However, after another review of the files, I found a few more issues that need to be addressed. Please check the suggestions and incorporate them. Also, as we discussed on knox messenger, it was decided to align the topic (Navigation of screens) with Animation, Components, Layer, and Layouts such as:
------> ##### Screen Navigation (Rename from 'Navigation of screens') <-------
------> ###### Navigator <-------
------> ###### ContentPage <-------
------> ###### DialogPage <-------

Please add this topic (Screen Navigation) on line 264 and proceed below in the same order shown above.

Finally, the overview file of NUI (/application/dotnet/guides/user-interface/nui/overview/) also needs to be updated to add a short summary of the new topic (Screen Navigation)

jaehyun0cho commented:
> Also, as we discussed on knox messenger, it was decided to align the topic (Navigation of screens) with Animation, Components, Layer, and Layouts such as: ------> ##### Screen Navigation (Rename from 'Navigation of screens') <------- ------> ###### Navigator <------- ------> ###### ContentPage <------- ------> ###### DialogPage <-------
> 
> Please add this topic (Screen Navigation) on line 264 and proceed below in the same order shown above.
> 
> Finally, the overview file of NUI (/application/dotnet/guides/user-interface/nui/overview/) also needs to be updated to add a short summary of the new topic (Screen Navigation)

@omar-srbd Thank you for your review. About the position of "Screen Navigation", I thought that changing "Overview" to "Navigator" is enough. Is it ok if I put "Screen Navigation" with the same level of "Layouts" even though "Navigator", "ContentPage", "DialogPage" are actually under NUI Components?

jaehyun0cho commented:
@omar-srbd 
I have updated the structure of this PR as follows.
nui/nui-components/navigation -> nui/nui-components/page-navigation
nui/nui-components/navigation/Navigator.md -> nui/nui-components/page-navigation/page-navigation.md

Also I changed the topic name from "Screen Navigation" to "Page Navigation".

Because page-navigation.md describes the overview of Page Navigation by explaining Tizen.NUI.Components.Page and Tizen.NUI.Components.Navigator.
Consequently, this change leads the topic name changes as well.

omar-srbd commented on docs/application/dotnet/guides/user-interface/nui/nui-components/page-navigation/page-navigation.md:
```diff
@@ -0,0 +1,31 @@
+
+## Overview
+
+NUI provides a system to organize screens and navigate between the screens.
```
```suggestion
NUI provides a system to organize screens and navigate between them.
```

omar-srbd commented on docs/application/dotnet/guides/user-interface/nui/overview.md:
```diff
@@ -29,6 +29,10 @@ NUI offers the following main features:
 
+- [Screen Navigation](nui-components/navigation/Navigator.md)
+
+  Provides a system to organize screens and navigate between the screens.
```
```suggestion
  Provides a system to organize screens and navigate between them.
```

omar-srbd commented on docs/application/dotnet/guides/user-interface/nui/nui-components/page-navigation/DialogPage.md:
```diff
@@ -0,0 +1,157 @@
+<div id="TabSection1">
+    <div class="sampletab " id="ProjectCreateTab">
+        <button id="DialogPage-Example-CSharp" class="tablinks " onclick="openTabSection(event, 'DialogPage-Example-CSharp', 'TabSection1') ">C#</button>
+        <button id="DialogPage-Example-Xaml" class="tablinks " onclick="openTabSection(event, 'DialogPage-Example-Xaml', 'TabSection1') ">XAML</button>
```
```suggestion
        <button id="DialogPage-Example-Xaml" class="tablinks " onclick="openTabSection(event, 'DialogPage-Example-Xaml', 'TabSection1') ">Xaml</button>
```

omar-srbd commented on docs/application/dotnet/guides/user-interface/nui/nui-components/page-navigation/ContentPage.md:
```diff
@@ -0,0 +1,130 @@
+<div id="TabSection1">
+    <div class="sampletab " id="ProjectCreateTab">
+        <button id="ContentPage-Example-CSharp" class="tablinks " onclick="openTabSection(event, 'ContentPage-Example-CSharp', 'TabSection1') ">C#</button>
+        <button id="ContentPage-Example-Xaml" class="tablinks " onclick="openTabSection(event, 'ContentPage-Example-Xaml', 'TabSection1') ">XAML</button>
```
```suggestion
        <button id="ContentPage-Example-Xaml" class="tablinks " onclick="openTabSection(event, 'ContentPage-Example-Xaml', 'TabSection1') ">Xaml</button>
```

omar-srbd commented on docs/application/dotnet/guides/user-interface/nui/overview.md:
```diff
@@ -29,6 +29,10 @@ NUI offers the following main features:
 
   Provides advanced reusable laying out capabilities.
 
+- [Screen Navigation](nui-components/navigation/Navigator.md)
```
```suggestion
- [Page Navigation](nui-components/page-navigation/page-navigation.md)
```

omar-srbd reviewed (commented):
@jaehyun0cho, I have a few more suggestions.
Please check and incorporate them accordingly.

Regarding your https://github.com/Samsung/tizen-docs/pull/1745#issuecomment-1368614332, the arrangement of the files seems alright to me. It is better to keep them as they are now.

jaehyun0cho commented:
@omar-srbd Thank you for your review. I have updated as you suggested.

omar-srbd reviewed (commented):
@jaehyun0cho, thanks for your most recent commit.
However, after one final check on the topic arrangements, it seems there is another change to be made.
Please check the suggestion for your reference.

omar-srbd commented:
Hello @jaehyun0cho, this PR has been inactive for a while.
Please check the feedback https://github.com/Samsung/tizen-docs/pull/1745#pullrequestreview-1233712861 and incorporate the change.
Thank you.

jaehyun0cho commented on docs/application/dotnet/guides/user-interface/nui/overview.md:
```diff
@@ -29,6 +29,10 @@ NUI offers the following main features:
 
   Provides advanced reusable laying out capabilities.
 
+- [Page Navigation](nui-components/page-navigation/page-navigation.md)
+
+  Provides a system to organize screens and navigate between them.
+
```
Thank you for your comment.
However, I think that it would be better keeping the current organization.
Because the components for page navigation are different from other components. Because they are not simple components but they are containers to contain other components and show navigations between them.
So alghough the components for page navigation is under the namespace of NUI.Components but I think that they should be described in the overview of NUI section (like the current version).

omar-srbd replied:
@jaehyun0cho, thanks for clarifying the matter. I will proceed to the next stage with the current topic arrangement.
